### PR TITLE
refactor: return `204` instead of `200` for `PUT /authenticator/:id`

### PR DESCRIPTION
**Pull request by sstrigler**
Fixes [EMQX-7982](https://emqx.atlassian.net/browse/EMQX-7982)

Repost of https://github.com/emqx/emqx/pull/9434 after it has been reverted. 

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/` dir
- [x] For internal contributor: there is a jira ticket to track this change

### CODE REVIEW

**Review by lafirest — DISMISSED** [review]

File: apps/emqx_authn/src/emqx_authn_api.erl [hunk 1]
```diff
@@ -1006,11 +1000,10 @@ update_authenticator(ConfKeyPath, ChainName, AuthenticatorID, Config) ->
         )
     of
         {ok, #{
-            post_config_update := #{emqx_authentication := #{id := ID}},
-            raw_config := AuthenticatorsConfig
+            post_config_update := #{emqx_authentication := #{id := _ID}},
```
Comment (lafirest) [inline apps/emqx_authn/src/emqx_authn_api.erl:1003]: nit: these two lines can be deleted

File: .github/workflows/run_jmeter_tests.yaml [hunk 2]
```diff
@@ -190,8 +190,8 @@ jobs:
         echo PGSQL_IP=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' pgsql-tls) >> $GITHUB_ENV
     - uses: actions/checkout@v3
       with:
-        repository: emqx/emqx-fvt
-        ref: broker-autotest
+        repository: sstrigler/emqx-fvt
```
Comment (lafirest) [inline .github/workflows/run_jmeter_tests.yaml:193]: why make this change

Comment (sstrigler) [reply .github/workflows/run_jmeter_tests.yaml:193]: Sorry, I should have hidden this PR, it's not ready. There are problems with the JMeter integration tests suite and I'm about to fix them. There'll be a series of PRs before we can merge this one.

File: apps/emqx_authn/src/emqx_authn_api.erl (hunk 1 above)
Comment (sstrigler) [reply apps/emqx_authn/src/emqx_authn_api.erl:1003]: Thought I'd keep it as a sort of sanity check. But yeah, just looking at `ok` should do it.

**Comment by id** [comment]
@sstrigler please rebase on latest master

**Review by zmstone — APPROVED** [review]

**Pull request merged**